api/types.go

```diff
@@ -141,27 +142,55 @@ type SeriesType string
 // end =   0 mod resolution
 //
 // This range is inclusive of Start and End (i.e. [Start, End]). Start and End
-// are Unix second timestamps. Resolution is in seconds.
-// TODO: Make these ms
+// are Unix milliseconds timestamps. Resolution is in milliseconds.
+// (Millisecond precision allows an effective range of 290 million years in each direction)
 type Timerange struct {
-	Start      int64
-	End        int64
-	Resolution int64
+	start      int64
+	end        int64
+	resolution int64
+}
+
+// DefaultTimerange is a valid timerange which can be safely used
+func DefaultTimerange() Timerange {
```

Comment:
The original reason is that the timerange `{0, 0, 0}` is invalid, so any code that wanted to build a valid (but placeholder/useless) `Timerange` would be able to do so without having to check errors or discard them.

However, I did just realize that there are some issues elsewhere that need to be addressed with this change, namely in the contextevaluation generation.